presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -430,23 +429,11 @@ public ActualProperties visitJoin(JoinNode node, List<ActualProperties> inputPro
                             .unordered(true)
                             .build();
                 case FULL:
-                    if (probeProperties.getNodePartitioning().isPresent()) {
-                        Partitioning nodePartitioning = probeProperties.getNodePartitioning().get();
-                        ImmutableList.Builder<Expression> coalesceExpressions = ImmutableList.builder();
-                        for (Symbol column : nodePartitioning.getColumns()) {
-                            for (JoinNode.EquiJoinClause equality : node.getCriteria()) {
-                                if (equality.getLeft().equals(column) || equality.getRight().equals(column)) {
-                                    coalesceExpressions.add(new CoalesceExpression(ImmutableList.of(equality.getLeft().toSymbolReference(), equality.getRight().toSymbolReference())));
-                                }
-                            }
-                        }
-
-                        return ActualProperties.builder()
-                                .global(partitionedOn(Partitioning.createWithExpressions(nodePartitioning.getHandle(), coalesceExpressions.build()), Optional.empty()))
-                                .build();
-                    }
+                    // We can't say anything about the partitioning scheme because any partition of
+                    // a hash-partitioned join can produce nulls in case of a lack of matches
                     return ActualProperties.builder()
                             .global(probeProperties.isSingleNode() ? singleStreamPartition() : arbitraryPartition())
+                            .unordered(unordered)
```

Comment:
I think you can remove this as this is noop

Reply:
I believe it was there before these changes were introduced. I just reverted this chunk of code.

Reply:
it was, but it's still a noop (I removed similar line in this method).